### PR TITLE
fix: add separate workflow for CEL policies to unblock e2e tests

### PR DESCRIPTION
**Pull request by Chandan-DK**
## Description
This PR resolves `Strategy expansion exceeded 256 results for job 'chainsaw'` error that is currently blocking the E2E tests from running. 

A separate workflow file has been created for CEL policies and this PR makes use of composite actions to reuse steps in both the workflow files (`test.yml` and `cel-test.yml`)

## Checklist

- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Comment by Chandan-DK** [comment]
Made this PR by mistake. But if it

**Comment by chipzoller** [comment]
Yes, I think this would be a good approach! Please do reopen!

**Comment by realshuting** [comment]
Hi @Chandan-DK - let us know if you are still interested in contribution!

**Comment by realshuting** [comment]
Thanks @Chandan-DK !

All new tests failed due to this:
>        === ERROR
        admission webhook "validate-policy.kyverno.svc" denied the request: no unique match for kind GitRepository
    l.go:53: | 18:25:20 | verify-git-repositories | 01 - Create policy and verify  | CREATE    | ERROR | kyverno.io/v1/ClusterPolicy @ verify-git-repositories
        === ERROR
        admission webhook "validate-policy.kyverno.svc" denied the request: no unique match for kind GitRepository

Looks like something is missing, can you take a look?

File: .github/actions/setup-env/action.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - name: Wait for kyverno ready
+      shell: bash
+      run: |
+        set -e
+        kubectl wait --namespace kyverno --for=condition=ready pod --selector '!job-name' --timeout=60s
+    - name: Install CRDs
+      shell: bash
+      run: |
+        set -e
+        kubectl apply -f ./.chainsaw/crds
```
Comment (chipzoller) [inline .github/actions/setup-env/action.yaml:51]: This has to point to the CRDs folder for all tests.

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: .github/actions/setup-env/action.yaml (hunk 1 above)
Comment (Chandan-DK) [reply .github/actions/setup-env/action.yaml:51]: It is pointing to [this](https://github.com/kyverno/policies/tree/main/.chainsaw/crds) folder. We can see that the CRDs are being created when running this command. https://github.com/kyverno/policies/actions/runs/10163856510/job/28107824800#step:3:363

**Comment by Chandan-DK** [comment]
> Thanks @Chandan-DK !
> 
> All new tests failed due to this:
> 
> > ```
> >    === ERROR
> > ```
> 
> ```
>     admission webhook "validate-policy.kyverno.svc" denied the request: no unique match for kind GitRepository
> l.go:53: ���������| 18:25:20 | verify-git-repositories | 01 - Create policy and verify  | CREATE    | ERROR | kyverno.io/v1/ClusterPolicy @ verify-git-repositories
>     === ERROR
>     admission webhook "validate-policy.kyverno.svc" denied the request: no unique match for kind GitRepository
> ```
> 
> Looks like something is missing, can you take a look?

`verify-git-repositories` cel policy is not being created due to this issue (https://github.com/kyverno/kyverno/issues/10313). I have removed this policy for now.

File: .github/actions/setup-env/action.yaml (hunk 1 above)
Comment (chipzoller) [reply .github/actions/setup-env/action.yaml:51]: Ok, we can reintroduce later.

**Review by chipzoller — COMMENTED** [review]
Could we rename the `.chainsaw-test` directory of the problematic policy rather than just deleting it?

**Comment by Chandan-DK** [comment]
Sure, but there will still be noise in the CI since Chainsaw executes the `chainsaw-test.yaml` file. We could rename this file too, to prevent noise. What do you think?

**Comment by chipzoller** [comment]
I would ask @eddycharly what's the most reliable thing to do here. I'm not sure what the current recommended approach is for Chainsaw given its present state.

**Comment by Chandan-DK** [comment]
Chainsaw by default looks for the `chainsaw-test.yaml` file inside every folder. We could rename the `.chainsaw-test` directory of the problematic policy to something like `.chainsaw-test-rename-after-issue-10313-fix` to make it easy to identify this problematic policy in the future. Changing the directory name won't stop the `chainsaw-test.yaml` file that is present in it from being executed. So, we should probably name the file this way `.chainsaw-test-rename-after-issue-10313-fix.yaml` and avoid a failing chainsaw test for the time being.

By the way, the policies that have been deleted due to this issue are being tracked [here](https://github.com/kyverno/kyverno/issues/10313). I have been referencing the link of the issue in my commits when deleting policies. We could use those commits to help add the policies back when the issue is resolved.
image

**Comment by chipzoller** [comment]
Ok, that sounds good. Let's give it a try.

**Review by chipzoller — APPROVED** [review]
Much thanks! I'm going to go ahead and merge this so we can run our e2e tests again.

**Comment by realshuting** [comment]
Good job @Chandan-DK !

**Comment by Chandan-DK** [comment]
Thanks @realshuting @chipzoller !

**Pull request merged**